lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1982,6 +2002,35 @@ internal class EvaluatingCompiler(
             CompiledGroupByItem(alias.text.exprValue(), uniqueName, compileAstExpr(it.expr))
         }
 
+    private fun compileOrderByExpression(sortSpecs: List<PartiqlAst.SortSpec>): List<CompiledOrderByItem> =
+        sortSpecs.map {
+            it.orderingSpec
+                ?: errNoContext(
+                    "SortSpec.orderingSpec was not specified",
+                    errorCode = ErrorCode.INTERNAL_ERROR,
+                    internal = true
+                )
+
+            it.nulls
+                ?: errNoContext(
+                    "SortSpec.nulls was not specified",
+                    errorCode = ErrorCode.INTERNAL_ERROR,
+                    internal = true
+                )
+
+            val orderSpec = when (it.orderingSpec) {
+                is PartiqlAst.OrderingSpec.Asc -> OrderingType.ASC
+                is PartiqlAst.OrderingSpec.Desc -> OrderingType.DESC
+            }
+
+            val nullsSpec = when (it.nulls) {
+                is PartiqlAst.NullsSpec.NullsFirst -> NullsType.FIRST
+                is PartiqlAst.NullsSpec.NullsLast -> NullsType.LAST
+            }
+
+            CompiledOrderByItem(orderSpec, nullsSpec, compileAstExpr(it.expr))
+        }
```

Comment:
I think [NaturalExprValueComparators](https://github.com/partiql/partiql-lang-kotlin/blob/67e3adf452d1d411aaba71de2635e12bb25bca2e/lang/src/org/partiql/lang/eval/NaturalExprValueComparators.kt) can be used here with some slight modification to case on the `OrderingSpec`. Using that could remove the need for `CompareExtentions.kt` and provide the implementation for some of the type comparisons.

Also, `NaturalExprValueComparators` has a lot of tests in [NaturalExprValueComparatorsTest.kt](https://github.com/partiql/partiql-lang-kotlin/blob/67e3adf452d1d411aaba71de2635e12bb25bca2e/lang/test/org/partiql/lang/eval/NaturalExprValueComparatorsTest.kt), which can reduce the amount of tests needed to be written.